### PR TITLE
 persistent recommendations created date

### PR DESCRIPTION
**Pull request by dpensi**
# Description
makes created_at column of recommendation table persistent (doesn't get updated)

Fixes CCXDEV-7858

## Type of change

Please delete options that are not relevant.

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Testing steps
make before_commit

## Checklist
* [ ] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

**Review by tisnik — APPROVED** [review]
ok, but probably it would be better to put the code into its own function/method? WDYT?

File: storage/storage_test.go [hunk 1]
```diff
@@ -1149,17 +1149,22 @@ func TestDBStorageWriteRecommendationForClusterAlreadyStoredAndDeleted(t *testin
...
+	expects.ExpectQuery(`SELECT created_at FROM recommendation`).
+		WillReturnRows(expects.NewRows([]string{"created_at"}).AddRow(time.Time{})).
+		RowsWillBeClosed()
 	expects.ExpectExec("DELETE FROM recommendation").
 		WillReturnResult(driver.RowsAffected(3))
 	expects.ExpectExec("INSERT INTO recommendation").
 		WillReturnResult(driver.ResultNoRows)
 	expects.ExpectCommit()
+	expects.ExpectClose()
+	expects.ExpectClose()
```
Comment (juandspy) [inline storage/storage_test.go:1161]: why is it necessary to expectClose twice?

**Review by juandspy — APPROVED** [review]

File: storage/storage.go [hunk 2]
```diff
@@ -940,6 +940,44 @@ func (storage DBStorage) insertRecommendations(
...
+		return nil, err
+	}
+
+	if exists := creationTimeRows.Next(); exists {
+		var oldTime time.Time
+		if err := creationTimeRows.Scan(&oldTime); err != nil {
+			log.Error().Err(err).Msg("error scanning old time from recommendation")
+			return nil, err
+		}
+		newTime = types.Timestamp(oldTime.UTC().Format(time.RFC3339))
```
Comment (tisnik) [inline storage/storage.go:967]: `newTime := ` allows you to delete line #953

**Review by tisnik — APPROVED** [review]
LGTM

File: storage/storage_test.go (hunk 1 above)
Comment (dpensi) [reply storage/storage_test.go:1161]: it will fail if i remove one, and it will fail even if I don't call Close()

**Pull request merged**